### PR TITLE
Update to wgpu 0.17 & latest egui

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Depends on https://github.com/emilk/egui/pull/3253
   * allows for a bit nicer callback handling on our side. Still not amazing but quite a bit better I reckon (no more locks and additional ref counting on our side!). I was hoping to use the new, more flexible `shared_paint_callback_resources` to store `ViewBuilder`, but it can't be accessed without a handle to the eframe renderer.
   The end goal would be to get rid of re_renderer's `per_frame_data_helper`. This will likely be enabled with wgpu's "Arcanization" effort as this will eliminate `wgpu::RenderPass`'s lifetime dependency. In that case we could store a `Mutex<ViewBuilder>` on `ReRendererCallback` 🤔 
* For the forseeable future we'll have to enable `fragile-send-sync-non-atomic-wasm` on wgpu (luckily egui itself no longer has this restriction). The problem is that a lot of our infrastructure is built around assuming `Send`/`Sync` whereas in actuality webgpu rendering resources can't be shared accross threads easily.
* Had to (discover and) work around https://github.com/gfx-rs/naga/issues/2436 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/2980) (if applicable)
* [x] Test WebGPU build

- [PR Build Summary](https://build.rerun.io/pr/2980)
- [Docs preview](https://rerun.io/preview/pr%3Aandreas%2Fupdate-wgpu/docs)
- [Examples preview](https://rerun.io/preview/pr%3Aandreas%2Fupdate-wgpu/examples)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -151,14 +153,14 @@ debug = true
...
-emath = { git = "https://github.com/emilk/egui", rev = "dd417cfc1acdf7ac88f214015d4d8e7a21835d30" }
-epaint = { git = "https://github.com/emilk/egui", rev = "dd417cfc1acdf7ac88f214015d4d8e7a21835d30" }
+ecolor = { git = "https://github.com/emilk/egui", rev = "a08f179c" }
+eframe = { git = "https://github.com/emilk/egui", rev = "a08f179c" }
+egui-wgpu = { git = "https://github.com/emilk/egui", rev = "a08f179c" }
+egui-winit = { git = "https://github.com/emilk/egui", rev = "a08f179c" }
+egui = { git = "https://github.com/emilk/egui", rev = "a08f179c" }
+egui_extras = { git = "https://github.com/emilk/egui", rev = "a08f179c" }
+emath = { git = "https://github.com/emilk/egui", rev = "a08f179c" }
+epaint = { git = "https://github.com/emilk/egui", rev = "a08f179c" }
```
Comment (abey79) [inline Cargo.toml:163]: My new favourite commit is 3c4223c6b18dc2ed441fc5b0077117e614565067 :)

**Review by abey79 — COMMENTED** [review]
Very glad to have this! I'm hardly able to comment about the code, but I made a note of my new favourite egui commit :)

File: Cargo.toml (hunk 1 above)
Comment (Wumpf) [reply Cargo.toml:163]: I'd prefer b896d641c578c04603adf07a9350b8bfc3c7ed9f though :smile:

Comment (abey79) [reply Cargo.toml:163]: Maybe we can reach an agreement on that 🤝

Comment (Wumpf) [reply Cargo.toml]: actually no, we can't. Need an even newer one

**Comment by Wumpf** [comment]
Depends on https://github.com/emilk/egui/pull/3260 now

**Review by emilk — APPROVED** [review]

**Comment by Wumpf** [comment]
Last commit fixes a Chrome only bug that Naga doesn't catch yet, see https://github.com/gfx-rs/naga/issues/2439
With that this version runs fine with WebGPU & Chrome again :)

**Comment by Wumpf** [comment]
(note that this fix was unrelated to the update here; this is rather keeping up with Chrome / the spec)

**Comment by Wumpf** [comment]
ignoring failed pr body update ci

**Pull request merged**